feature/experimental/p4rt/ate_tests/performance_test/performance_test.go

```diff
@@ -0,0 +1,955 @@
+	streamName                              = "p4rt"
+	gdpInLayers         layers.EthernetType = 0x6007
+	lldpInLayers        layers.EthernetType = 0x88cc
+	lldpEthertype                           = uint32(0x88cc)
+	gdpEtherType                            = uint32(0x6007)
+	deviceID                                = *ygot.Uint64(1)
+	portID                                  = *ygot.Uint32(10)
+	electionID                              = *ygot.Uint64(100)
+	metadataIngressPort                     = *ygot.Uint32(1)
+	metadataEgressPort                      = *ygot.Uint32(2)
+
+	Duration        = uint32(20)
+	gdpBitRate      = uint64(200000)
+	lldpBitRate     = uint64(100000)
+	trPacketRate    = uint64(324)
+	packetinPktsize = uint32(300)
+	srcMac          = flag.String("srcMac", "00:01:00:02:00:03", "source MAC address for PacketIn")
+	lldpDstmac      = flag.String("lldpDstmac", "01:80:c2:00:00:0e", "source MAC address for PacketIn")
+	gdpDstmac       = flag.String("gdpDstmac", "00:0a:da:f0:f0:f0", "source MAC address for PacketIn")
+	TTL1            = uint8(1)
+	HopLimit1       = uint8(1)
+	packetoutWait   = uint32(77) // Wait for the ATE traffic start, so both packetin and packetout hits DUT simultaneously.
+	ipv4PrefixLen   = uint8(30)
+	ipv6PrefixLen   = uint8(126)
+	packetCount     = 2000
+)
+
+var (
+	dutPort1 = attrs.Attributes{
+		Desc:    "dutPort1",
+		IPv4:    "192.0.2.1",
+		IPv6:    "2001:db8::1",
+		IPv4Len: ipv4PrefixLen,
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	atePort1 = attrs.Attributes{
+		Name:    "atePort1",
+		IPv4:    "192.0.2.2",
+		IPv6:    "2001:db8::2",
+		IPv4Len: ipv4PrefixLen,
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	dutPort2 = attrs.Attributes{
+		Desc:    "dutPort2",
+		IPv4:    "192.0.2.5",
+		IPv6:    "2001:db8::5",
+		IPv4Len: ipv4PrefixLen,
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	atePort2 = attrs.Attributes{
+		Name:    "atePort2",
+		IPv4:    "192.0.2.6",
+		IPv6:    "2001:db8::6",
+		IPv4Len: ipv4PrefixLen,
+		IPv6Len: ipv6PrefixLen,
+	}
+)
+
+type testArgs struct {
+	ctx          context.Context
+	leader       *p4rt_client.P4RTClient
+	dut          *ondatra.DUTDevice
+	ate          *ondatra.ATEDevice
+	top          *ondatra.ATETopology
+	gdppacketIO  PacketIO
+	lldppacketIO PacketIO
+	trpacketIO   PacketIO
+}
+
+type PacketIOPacket struct {
+	TTL            *uint8
+	SrcMAC, DstMAC *string
+	EthernetType   *uint32
+	HopLimit       *uint8
+}
+
+type PacketIO interface {
+	GetPacketTemplate() *PacketIOPacket
+	GetEgressPort() string
+	GetIngressPort() string
+}
+
+type LLDPPacketIO struct {
+	PacketIOPacket
+	IngressPort string
+}
+
+type GDPPacketIO struct {
+	PacketIOPacket
+	IngressPort string
+}
+
+type TraceroutePacketIO struct {
+	PacketIOPacket
+	IngressPort string
+	EgressPort  string
+}
+
+// programmTableEntry programs or deletes p4rt table entry based on delete flag.
+func programmTableEntry(ctx context.Context, t *testing.T, client *p4rt_client.P4RTClient, delete bool) error {
+	t.Helper()
```

Comment:
There is no need to accept a t parameter or call t.Helper if t isn't otherwise used in the function.